### PR TITLE
Add player join and quit messages.

### LINKED ISSUE #188: Add join and quit messages
- [x] I've searched existing issues and couldn't find a duplicate.

**Is your feature request related to a problem? Please describe.**
- Pumpkin currently doesn't have join and quit messages.

**Describe the solution you'd like**
- Join message
  - Will position in `client_packets.rs`, at the top of `finish_login`, above where we send them the success packet.
- Quit message
  - I can't find where to put this, please could a maintainer advise
- Kick log message
  - Only an INFO log message to note that a player was kicked.

**To-do list**
- [x] Maintainers approve idea
- [x] Contributor indicates good place to handle the quit message code; edit above
- [x] Started work on a PR (linked to issue).

### PR DESCRIPTION
**Pull request by lokka30**
This implementation will send a 'joined the game' and 'left the game' message akin to the vanilla server when a player joins or disconnects respectively.

One caveat with this implementation is that it only broadcasts the join/quit messages to players in the same world, unlike the vanilla server, which broadcasts it to all online players (regardless of their world).

I discussed this with Alex who suggested to keep it as per-world (seemingly for now?) - [Discord conversation](https://discord.com/channels/1268592337445978193/1268611318617866261/1299668206734147597).

Regardless, this implementation works perfectly on my end. (yes, it has been tested)

Resolves #188 

Screenshot:
image

- [x] Ready for review/merge

### CODE REVIEW

**Comment by lokka30** [comment]
I'd like to add some console logging in before we merge actually.

**Comment by lokka30** [comment]
image
Done, ready now assuming all compiles well 😛

**Comment by Snowiiii** [comment]
I would prefer to have the message in `world` not in player

**Comment by lokka30** [comment]
Thanks for the review @Snowiiii, I've moved disconnect message logic from `player` to `world` in 7c06b7a.

I think this causes implications though. Wouldn't this send players in the world a 'left the game' message when a player is removed from the world, even if they're on the same server, perhaps moving to another world?
If so, this would also cause the disconnect message to be a per-world thing (non-vanilla behaviour), when the join message is a per-server thing (vanilla behaviour).

Please let me know what you think

File: pumpkin/src/server/mod.rs [hunk 1]
```diff
@@ -113,6 +116,15 @@ impl Server {
...
             }
         }
+
+        // Handle join message
+        let msg_txt = format!("{} joined the game.", player.gameprofile.name.as_str());
+        let msg_comp = TextComponent::text(msg_txt.as_str()).color_named(NamedColor::Yellow);
+        for player in world.current_players.lock().await.values() {
+            player.send_system_message(&msg_comp).await;
+        }
+        log::info!("{}", msg_comp.to_pretty_console());
```
Comment (Snowiiii) [inline pumpkin/src/server/mod.rs:126]: Let put this then into world as well

Comment (lokka30) [reply pumpkin/src/server/mod.rs]: Resolved in b5e01db 🙂 
Do we plan on mimicking vanilla behavior (per-server join/quit rather than per-world) in future?

Comment (Snowiiii) [reply pumpkin/src/server/mod.rs]: Not sure, We could make it  a config setting, but it would be annoying. For now i would leave it per world

**Comment by Snowiiii** [comment]
Looks good now.
Thanks @lokka30

**Pull request merged**